test/src/test/java/org/corfudb/runtime/collections/CorfuTableTest.java

```diff
@@ -245,4 +254,46 @@ public void snapshotInvariant() {
         final Stream<Map.Entry<Integer, Integer>> result = map.entryStream();
         result.forEach(e -> map.put(new Random().nextInt(), 0));
     }
+
+    @Test
+    @SuppressWarnings({"unchecked", "checkstyle:magicnumber"})
+    public void canHandleHoleInTail() {
+        UUID streamID = UUID.randomUUID();
+
+        CorfuTable<String, String> corfuTable = getDefaultRuntime()
+                .getObjectsView()
+                .build()
+                .setTypeToken(new TypeToken<CorfuTable<String, String>>() {})
+                .setStreamID(streamID)
+                .open();
+
+        corfuTable.put("k1", "dog fox cat");
+        corfuTable.put("k2", "dog bat");
+        corfuTable.put("k3", "fox");
+
+        // create a hole
+        TokenResponse tokenResponse =  getDefaultRuntime()
+                .getSequencerView()
+                .next(streamID);
+
+        Token token = tokenResponse.getToken();
+
+        getDefaultRuntime().getAddressSpaceView()
```

Comment:
Can you read back the address and make sure its a hole?

Reply:
Done.